### PR TITLE
Create SNS Producer

### PR DESCRIPTION
**Pull request by jleber**
# ✨ New Feature

This MR contains the inclusion of SNS producer and a docker-compose file that provides localstack and aws-cli as services to create all infra stuff to simulate the usage of SNS and so on.

# :repeat: Steps to reproduce

Run the following commands at the project root, on your terminal:

```

make build-app
make build-app-infra

export AWS_ENDPOINT="http://localhost:4566"
export AWS_ACCESS_KEY_ID=foo
export AWS_SECRET_ACCESS_KEY=bar

irb

require 'pipefy_message'
message = PipefyMessage::Test.new
message.hello
```

# 🗂 Related cards

[[async] Create SNS producer inside the gem](https://app.pipefy.com/open-cards/490520520)

### CODE REVIEW

**Review by viniciuscbm — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
```
Comment (viniciuscbm) [inline Makefile:1]: Delete first line :nail_care:

File: lib/pipefy_message/broker/aws/sns/publisher.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "aws-sdk-sns"
+require "json"
+
+module PipefyMessage
+  # Aws SNS Publisher class to publish json messages into a specific topic
+  class SnsPublisher
+    def initialize
+      aws_config = PipefyMessage::AwsProviderConfig.instance
```
Comment (viniciuscbm) [inline lib/pipefy_message/broker/aws/sns/publisher.rb:10]: We could run like this here?

`PipefyMessage::AwsProviderConfig.instance.setup_connection`

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        sleep 10
+
+        # Creating SNS Topics
+        aws --endpoint-url=http://localstack:4566 sns create-topic --name pipefy-local-topic
+
+        # Creating SQS Queues
+        aws --endpoint-url=http://localstack:4566 sqs create-queue --queue-name pipefy-local-queue
+
+        # Creating Subscription from SQS to SNS
+        aws --endpoint-url=http://localstack:4566 sns subscribe --topic-arn arn:aws:sns:us-east-1:000000000000:pipefy-local-topic --protocol sqs --notification-endpoint http://localstack:4566/000000000000/pipefy-local-queue
```
Comment (looslidias) [inline docker-compose.yml:36]: Double checking: I think this ARNs will not change, as they may be the default for `localstack`. Is that so?  😄

**Review by viniciuscbm — APPROVED** [review]

**Review by looslidias — APPROVED** [review]
As we are just delivering a simple publisher/subscriber, sounds fine to me!

Later time, after properly adding the abstraction, we may come with better log structure, services and etc.

File: docker-compose.yml (hunk 3 above)
Comment (jleber) [reply docker-compose.yml:36]: yes, you are correct for localstack the ARN is always the same

**Comment by jleber** [comment]
> As we are just delivering a simple publisher/subscriber, sounds fine to me!
> 
> Later time, after properly adding the abstraction, we may come with better log structure, services and etc.

Probably the abstraction will make us change some points at the consumer and publisher but for now, the goal is only to make a simple and useful integration for SNS and SQS(on another task)

**Pull request merged**